### PR TITLE
resolve #210

### LINKED ISSUE #210: DataRepo: Better search results
When I type in the exact symbol, I would expect that stock to come first in the search results. This is not happening at the moment. The searchByPrefix function in the DataRepo would have to be updated to improve this. Ideally we would find all exact matches and show them first. I am not sure if going through the whole list of sonifiables everytime wouldn't be too slow, so potentially we would need a sorted list of symbols (maybe packing them even to prevent traversing a list of pointers). The performance part of this issue would be rather optional though, I think
grafik

### PR DESCRIPTION
**Pull request by ArtInLines**
Exact matches are now shown first.

@MalteRichert if you wanna see some really messy code, look at my first commit here xd

### CODE REVIEW

**Review by nichtLehdev — COMMENTED** [review]
Looks good to me. Couldn't test tho.

**Review by MalteRichert — APPROVED** [review]
I think it works. One thing though: Is there a reason why we only search from the beginning right now? I find it way more intuitive to search anywhere in the string so that we could actually return Siemens Healthineers for a search term "Healthineers". @ArtInLines do you know what I mean?

File: src/main/java/dhbw/si/dataRepo/DataRepo.java [hunk 1]
```diff
@@ -102,39 +102,48 @@ public static void init() throws AppError {
...
+			// allocating new strings in each iteration, we should just store names and symbols as
+			// lowercase in some array, that we can traverse then
+			String name = s.name.toLowerCase();
+			String symbol = s.getId().symbol.toLowerCase();
+
+			if (name.equals(prefix) || symbol.equals(prefix)) {
+				exactResults.add(s);
+				if ((usedLen++) == totalLen) // usedLen++ is wrapped in parans, bc I'm not sure about the order of operations with ++ in java
+					break;
+			} else if (s != null && (name.startsWith(prefix) || symbol.startsWith(prefix))) {
```
[the remark this answers is not in the document]
Comment (ArtInLines) [reply src/main/java/dhbw/si/dataRepo/DataRepo.java:141]: That does make a lot more sense lol. No idea why I never thought about that. I think for symbols it makes sense to only search at the beginning of the string, but you are right, that contains would make more sense for names. I'll change that in a moment

**Comment by ArtInLines** [comment]
I am currently incorporating this and some other changes into #214 so I will close this pr here and bring the changes to there

**Pull request closed without merging**